### PR TITLE
Add optional event_ids to get_participation_overview()

### LINKED ISSUE #1824: [Task]: Add an optional parameter of eventIds to filter participation overviews
## Description

Whenever we request the participation overview of an account, we get back all the outputs related  to governance. It takes a long time to create the whole participation object to filter it again for a given ID.

## Motivation

Firefly would like to optimize their loading speeds and API requests

## Requirements

Write a list of what you want this feature to do.

1. add an optional parameter to `get_participation_overview` that represents an array of event ids
2. if the array is not present, return participation outputs for all events
3. otherwise return the outputs for the eventId's inside the array.

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add optional event_ids to get_participation_overview()
Add rocksdb feature to cli wallet
Fixed paths for documentation

## Links to any relevant issues

Fixes #1824 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

- Voting examples in nodejs with and without providing event ids to getParticipationOverview()
- CLI wallet voting and getting the participation overview without and with an event id

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by kwek20 — DISMISSED** [review]

File: cli/src/command/account.rs [hunk 1]
```diff
@@ -126,11 +126,14 @@ pub enum AccountCommand {
...
+    Vote {
+        event_id: ParticipationEventId,
+        answers: Vec<u8>,
+    },
     /// Stop participating to a given event
-    StopParticipating { event_id: String },
+    StopParticipating { event_id: ParticipationEventId },
     /// Calculate the participation overview of the account
-    ParticipationOverview,
+    ParticipationOverview { event_id: Option<ParticipationEventId> },
```
Comment (thibault-martinez) [inline cli/src/command/account.rs]: I think you can make it a Vec and clap should be able to deal with it...

Comment (Thoralf-M) [reply cli/src/command/account.rs]: Hm how to provide it then? Multiple events fail for me
```
participation overview 0x0344c97dc9cddc47f880fc1934e361636bf83029268f17faaac97c7be3865f7f 0x013d2ca528a25124231c720095859802f354f8760c0ff1301daba052d4419cf5
error: unrecognized subcommand
```

Comment (thibault-martinez) [reply cli/src/command/account.rs]: Maybe it doesn't like `Option<Vec<_>>`? I would argue that if we have a Vec, the Option is not needed in the CLI anyway

Comment (thibault-martinez) [reply cli/src/command/account.rs]: Assuming clap accepts empty vecs ofc

Comment (Thoralf-M) [reply cli/src/command/account.rs]: Yeah I tried first `Option<Vec<_>>`, but that didn't event accept a single one and then I tried `Vec<_>` and this seems only to work if I provide a single one
But usually one only cares about a single one anyways, no?

Comment (Thoralf-M) [reply cli/src/command/account.rs]: Ok got it working, was missing some clap arguments

File: cli/src/command/account.rs [hunk 2]
```diff
@@ -126,11 +126,17 @@ pub enum AccountCommand {
...
+        event_id: ParticipationEventId,
+        answers: Vec<u8>,
+    },
     /// Stop participating to a given event
-    StopParticipating { event_id: String },
+    StopParticipating { event_id: ParticipationEventId },
     /// Calculate the participation overview of the account
-    ParticipationOverview,
+    ParticipationOverview {
+        #[clap(short, long, value_parser, num_args = 1.., value_delimiter = ' ')]
```
Comment (thibault-martinez) [inline cli/src/command/account.rs:137]: We have a Vec for `Vote` and don't need this 🤔

Comment (thibault-martinez) [reply cli/src/command/account.rs:137]: But maybe it's because `ParticipationEventId` is more complex than `u8`

Comment (Thoralf-M) [reply cli/src/command/account.rs:137]: Yeah not sure what's the difference, but without it, it didn't work for me :shrug:

Comment (thibault-martinez) [reply cli/src/command/account.rs:137]: Alright no worries

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**